alembic/versions/2024_09_04_18dbadd8c436_rename_fastq_to_raw_data.py

```diff
@@ -0,0 +1,124 @@
+"""rename_fastq_to_raw_data
+
+Revision ID: 18dbadd8c436
+Revises: bb4c6dbad991
+Create Date: 2024-09-04 13:15:11.876822
+
+"""
+
+from enum import StrEnum
+
+import sqlalchemy as sa
+from sqlalchemy import orm, types
+from sqlalchemy.dialects import mysql
+from sqlalchemy.orm import DeclarativeBase, Mapped, mapped_column
+
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "18dbadd8c436"
+down_revision = "bb4c6dbad991"
+branch_labels = None
+depends_on = None
+
+
+base_options = (
+    "balsamic",
+    "balsamic-pon",
+    "balsamic-qc",
+    "balsamic-umi",
+    "demultiplex",
+    "fluffy",
+    "jasen",
+    "microsalt",
+    "mip-dna",
+    "mip-rna",
+    "mutant",
+    "raredisease",
+    "rnafusion",
+    "rsync",
+    "spring",
+    "taxprofiler",
+    "tomte",
+)
+old_options = sorted(base_options + ("fastq",))
+new_options = sorted(base_options + ("raw-data",))
+
+old_enum = mysql.ENUM(*old_options)
+new_enum = mysql.ENUM(*new_options)
+
+
+class Workflow(StrEnum):
+    BALSAMIC: str = "balsamic"
+    BALSAMIC_PON: str = "balsamic-pon"
+    BALSAMIC_QC: str = "balsamic-qc"
+    BALSAMIC_UMI: str = "balsamic-umi"
+    DEMULTIPLEX: str = "demultiplex"
+    FLUFFY: str = "fluffy"
+    JASEN: str = "jasen"
+    MICROSALT: str = "microsalt"
+    MIP_DNA: str = "mip-dna"
+    MIP_RNA: str = "mip-rna"
+    MUTANT: str = "mutant"
+    RAREDISEASE: str = "raredisease"
+    RAW_DATA: str = "raw-data"
+    RNAFUSION: str = "rnafusion"
+    RSYNC: str = "rsync"
+    SPRING: str = "spring"
+    TAXPROFILER: str = "taxprofiler"
+    TOMTE: str = "tomte"
+
+
+class Base(DeclarativeBase):
+    pass
+
+
+class Case(Base):
+    __tablename__ = "case"
+    id: Mapped[int] = mapped_column(primary_key=True)
+    data_analysis: Mapped[str | None] = mapped_column(
+        types.Enum(*(workflow.value for workflow in Workflow))
+    )
+
+
+class Analysis(Base):
+    __tablename__ = "analysis"
+
+    id: Mapped[int] = mapped_column(primary_key=True)
+    workflow: Mapped[str | None] = mapped_column(
+        types.Enum(*(workflow.value for workflow in Workflow))
+    )
+
+
+def upgrade():
+    bind = op.get_bind()
+    session = orm.Session(bind=bind)
+    try:
+        op.alter_column("case", "data_analysis", type_=new_enum)
+        session.query(Case).filter(Case.data_analysis == "fastq").update(
```

Comment:
If you alter the column to type `new_enum` you will get exceptions since there are cases with `data_analysis = fastq` in it, which does not work with the type. Same with analysis. You will probably need to do a migration which does 

1. Add option to have both `raw_data` and `fastq`.
2. Change all `fastq` to `raw_data`.
3. Remove option `fastq`.

Reply:
Thanks! Fixed